pontoon/sync/vcs/models.py

```diff
@@ -293,6 +295,33 @@ def resources_for_path(self, path):
                 if is_resource(filename):
                     yield os.path.join(root, filename)
 
+    def create_locale_directory(self, locale, path):
+        if not self.db_project.has_multi_locale_repositories:
+            source_directory = self.source_directory_path()
+            parent_directory = os.path.abspath(os.path.join(source_directory, os.pardir))
+            locale_directory = os.path.join(parent_directory, locale.code.replace('-', '_'))
+
+            # For asymmetric formats, create empty folder
+            if is_asymmetric_resource(path):
+                os.makedirs(locale_directory)
+
+            # For other formats, copy resources from source directory
+            else:
+                shutil.copytree(source_directory, locale_directory)
+
+                for root, dirnames, filenames in os.walk(locale_directory):
```

Comment:
We already have that case covered in https://github.com/mozilla/pontoon/pull/408/files/2c4f2d574d6510ebf76d4a76f2c1790384977eb0#diff-c3cfdee64b8608d7c2751b1df6d78acdR265.